### PR TITLE
clang tidy

### PR DESCRIPTION
**Pull request by kolay-v**

### CODE REVIEW

**Comment by LWSS** [comment]
This looks alright except for

When it removes the `(void)` in function calls. I want that void there because it means that I know the function has no arguments. When I don't know the arguments, I leave it `()`.

not a big fan of using auto but i'm

**Comment by kolay-v** [comment]
Ok, I'll disable this inspection. Also I have few questions. 
1. How I can level of enemy's skills?
2. Now I can get things like range attacks/stun of venga to dodge them?

**Comment by LWSS** [comment]
1/2 - I don't have those but what I would do to look into them is to print the entity list (home key), see the classname. Go into /tmp/dotanetvars.txt and look for netvars that seem like would work.

3 - `-insecure` if you want to be safe

File: src/SDK/panorama/IUIPanel.h [hunk 1]
```diff
@@ -73,14 +73,14 @@ namespace panorama
         virtual IUIPanel* FindChildInLayoutFile(const char *) = 0;
         virtual void unk11() = 0;
         virtual IUIPanel* FindPanelInLayoutFile(const char *) = 0;
-        virtual bool IsDecendantOf(IUIPanel const * const) = 0;
+        virtual bool IsDecendantOf(IUIPanel * const) = 0;
```
Comment (LWSS) [inline src/SDK/panorama/IUIPanel.h]: here it is transforming the const pointer to a const object into just a const pointer which might work but it should stay as it is.

File: src/SDK/CVScriptGameSystem.h [hunk 2]
```diff
@@ -7,8 +7,8 @@ class CVScriptGameSystem
     virtual void IGameSystem__Init(void) = 0;
 	virtual void IGameSystem__PostInit(void) = 0;
 	virtual void IGameSystem__Shutdown(void) = 0;
-	virtual void CVScriptGameSystem__GameInit(void* const EventGameInit_t) = 0;
-	virtual void CVScriptGameSystem__GameShutdown(void* const EventGameShutdown_t) = 0;
+	virtual void CVScriptGameSystem__GameInit(void* EventGameInit_t) = 0;
```
Comment (LWSS) [inline src/SDK/CVScriptGameSystem.h]: same thing here, I would like the const to stay

**Comment by LWSS** [comment]
what tool are you using to do this?

**Comment by kolay-v** [comment]
Clion

File: src/SDK/panorama/IUIPanel.h (hunk 1 above)
Comment (kolay-v) [reply src/SDK/panorama/IUIPanel.h]: Parameter 1 is const qualified in the function declaration; const qualification of parameters only have effect in function definition

File: src/McDota.cpp [hunk 3]
```diff
@@ -1,25 +1,21 @@
-#include <chrono>
-#include <fcntl.h>
-#include <sys/stat.h>
 #include <csignal>
-#include <fstream>
-#include <sstream>
-#include <zconf.h> //readlink
```
Comment (kolay-v) [inline src/McDota.cpp]: Should we include this?

File: src/SDK/CCollisionProperty.h [hunk 4]
```diff
@@ -11,20 +11,20 @@ struct ChangeAccessorFieldPathIndex_t;
 class CCollisionProperty
 {
 public:
-    virtual CSchemaClassBinding* Schema_DynamicBinding(void);
```
Comment (kolay-v) [inline src/SDK/CCollisionProperty.h]: Void

File: src/SDK/panorama/IUIPanel.h [hunk 5]
```diff
@@ -73,14 +73,14 @@ namespace panorama
...
         virtual void RemoveAndDeleteChildren(void) = 0;
         virtual void RemoveAndDeleteChildrenOfType(CPanoramaSymbol) = 0;
         virtual int GetChildCount(void) = 0;
         virtual IUIPanel* GetChild(int) = 0;
         virtual IUIPanel* GetFirstChild(void) = 0;
         virtual IUIPanel* GetLastChild(void) = 0;
-        virtual int GetChildIndex(IUIPanel const * const) = 0;
+        virtual int GetChildIndex(IUIPanel * const) = 0;
```
Comment (kolay-v) [inline src/SDK/panorama/IUIPanel.h]: Same const

File: src/SDK/panorama/IUIPanel.h [hunk 6]
```diff
@@ -163,7 +163,7 @@ namespace panorama
         virtual void AddClasses(CPanoramaSymbol *, unsigned int count) = 0;
         virtual void RemoveClass(const char *) = 0;
         virtual void RemoveClass(CPanoramaSymbol) = 0;
-        virtual void RemoveClasses(CPanoramaSymbol const * const, unsigned int) = 0;
+        virtual void RemoveClasses(CPanoramaSymbol * const, unsigned int) = 0;
         virtual void RemoveClasses(const char *) = 0;
```
Comment (kolay-v) [inline src/SDK/panorama/IUIPanel.h]: And there

File: src/SDK/IInputSystem.h [hunk 7]
```diff
@@ -560,7 +560,7 @@ class IInputSystem : public IAppSystem
 	virtual ButtonCode_t StringToButtonCode( const char *) = 0;
 	virtual AnalogCode_t StringToAnalogCode( const char *) = 0;
 
-	// Sleeps until input happens. Pass a negative number to sleep infinitely
+	// Sleeps until input happens. Pass a negative number to sleep infinitely TODO: Is it typo? (slee/sleep)
 	virtual void SleeUntilInput( int maxSleepTimeMS = -1 ) = 0;
```
Comment (LWSS) [inline src/SDK/IInputSystem.h]: lol

**Comment by LWSS** [comment]
I'll go ahead and accept this, first time seeing [[nodiscard]] is interesting.

**Pull request merged**